pkg/kapis/devops/v1alpha3/scm/scmhandler.go

```diff
@@ -48,14 +55,82 @@ func (h *handler) verify(request *restful.Request, response *restful.Response) {
 	secretNamespace := request.QueryParameter("secretNamespace")
 	server := common.GetQueryParameter(request, queryParameterServer)
 
-	_, code, err := h.getOrganizations(scm, server, secretName, secretNamespace, 1, 1, false)
+	code, err := 0, error(nil)
+	switch scm {
+	case "git":
+		if server == "" {
+			err := fmt.Errorf("server is required")
+			kapis.HandleError(request, response, err)
+			response.WriteHeaderAndEntity(http.StatusBadRequest, err)
+			return
+		}
+		code, err = h.checkRepoAccess(server, secretName, secretNamespace)
+
+	default:
+		_, code, err = h.getOrganizations(scm, server, secretName, secretNamespace, 1, 1, false)
+	}
 
 	response.Header().Set(restful.HEADER_ContentType, restful.MIME_JSON)
 	verifyResult := git.VerifyResult(err, code)
 	verifyResult.CredentialID = secretName
 	_ = response.WriteAsJson(verifyResult)
 }
 
+func (h *handler) checkRepoAccess(repourl, secretName, secretNamespace string) (int, error) {
+	user, token, privateKey, err := h.getTokenWithSShFromSecret(
+		context.Background(), &v1.SecretReference{
+			Namespace: secretNamespace, Name: secretName,
+		})
+	if err != nil {
+		return http.StatusOK, err
+	}
+
+	storage := memory.NewStorage()
+	remote := gogit.NewRemote(storage, &config.RemoteConfig{
+		Name: "origin",
+		URLs: []string{repourl},
+	})
+
+	auth, err := getAuthMethod(repourl, user, token, privateKey)
+	if err != nil {
+		return http.StatusBadRequest, err
+	}
+
+	_, err = remote.List(&gogit.ListOptions{Auth: auth})
+	if err != nil {
+		return http.StatusForbidden, handleGitError(err)
+	}
+
+	return http.StatusOK, nil
+}
+
+func (h *handler) getTokenWithSShFromSecret(ctx context.Context, ref *v1.SecretReference) (username, token string, privateKey []byte, err error) {
```

Comment:
why not modify the existing [getTokenFromSecret](https://github.com/kubesphere/ks-devops/blob/0c9f4b18f5cd5730275a7a4231585f3b0f1ba594/pkg/client/git/client.go#L76) or leverage it?

Reply:
done